### PR TITLE
test: more incentive test flows

### PR DESCRIPTION
**Pull request by toteki**

### CODE REVIEW

File: x/incentive/keeper/scenario_test.go [hunk 1]
```diff
@@ -216,23 +216,140 @@ func TestZeroBonded(t *testing.T) {
...
+	// actually claim the rewards (same amount)
+	rewards, err = k.UpdateAccount(k.ctx, alice)
+	require.NoError(k.t, err)
+	require.Equal(
+		k.t,
+		sdk.NewCoins(sdk.NewInt64Coin(umee, 10_000000)),
+		rewards,
+		"alice claimed rewards at time 220",
+	)
+}
```
[the remark this answers is not in the document]
Comment (toteki) [reply x/incentive/keeper/scenario_test.go:277]: added

File: x/incentive/keeper/scenario_test.go [hunk 2]
```diff
@@ -216,23 +216,140 @@ func TestZeroBonded(t *testing.T) {
 	)
 
 	// In this test case, an incentive program is started but no uTokens of the incentivized denom are
-	// bonded during its first half of runtime. during this time, it must not distribute rewards.
+	// bonded during its first half of runtime. During this time, it must not distribute rewards.
 	// During the remaining half of the program, all rewards must be distributed (spread evenly over
 	// the remaining time.)
```
[the remark this answers is not in the document]
Comment (toteki) [reply x/incentive/keeper/scenario_test.go]: moved these descriptions to the godoc of each test scenario

File: x/incentive/keeper/scenario_test.go [hunk 3]
```diff
@@ -171,68 +178,325 @@ func TestBasicIncentivePrograms(t *testing.T) {
...
+	k.mustBeginUnbond(alice, coin.New(uUmee, 100_000000))
+
+	// complete the program
+	k.advanceTimeTo(programStart + 100) // 100% duration
+	program = k.getProgram(1)
+	require.Equal(t, incentive.ProgramStatusCompleted, k.programStatus(1), "program 1 status completed (time 200)")
+	require.Equal(t, sdk.NewInt(3_333334), program.RemainingRewards.Amount, "two thirds of program rewards distributed")
+
+	// measure pending rewards (zero)
+	noRewards := sdk.NewCoins()
```
[the remark this answers is not in the document]
Comment (toteki) [reply x/incentive/keeper/scenario_test.go:328]: There is no additional brevity gained by switching those two expressions, and since this is package `keeper` and not `keeper_test`, I would prefer not to pollute the namespace

**Comment by toteki** [comment]
Added the final test scenario

There is also documentation (in the EndBlocker functions) about what happens in the zero-bonded edge case, which I have highlighted in the diff by adding more.

File: x/incentive/keeper/scenario_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (toteki) [reply x/incentive/keeper/scenario_test.go:326]: added as zeroCoins

File: x/incentive/README.md [hunk 4]
```diff
@@ -67,6 +67,12 @@ Reward distribution math is
 
 When multiple incentive programs are active simultaneously, they compute their rewards independently.
 
+Additionally, when no users are eligible for an incentive program's rewards while it is active, it refrains from distributing any rewards until eligible bonds exist.
```
[the remark this answers is not in the document]
Comment (toteki) [reply x/incentive/README.md]: updated

File: x/incentive/keeper/scenario_test.go [hunk 5]
```diff
@@ -171,68 +178,391 @@ func TestBasicIncentivePrograms(t *testing.T) {
...
+func TestRejoinScenario(t *testing.T) {
+	t.Parallel()
+	k := newTestKeeper(t)
+	k.initCommunityFund(
+		coin.New(umee, 1000_000000),
+	)
+
+	// an incentive program with 10 UMEE rewards will run from t=100 to t=200
+	programStart := int64(100)
+	k.addIncentiveProgram(uUmee, programStart, 100, sdk.NewInt64Coin(umee, 10_000000), true)
```
[the remark this answers is not in the document]
Comment (toteki) [reply x/incentive/keeper/scenario_test.go]: done

**Comment by toteki** [comment]
Resolved all issues, but need to redo PR due to unsigned commits

https://github.com/umee-network/umee/pull/2141
